sql/src/main/java/io/crate/planner/projection/MergeProjection.java

```diff
@@ -0,0 +1,154 @@
+        this.outputs = outputs;
+        this.orderBy = orderBy;
+        this.reverseFlags = reverseFlags;
+        this.nullsFirst = nullsFirst;
+        assert this.orderBy.size() == this.reverseFlags.length : "reverse flags length does not match orderBy items count";
+        assert this.nullsFirst.length == this.reverseFlags.length;
+    }
+
+    @Override
+    public ProjectionType projectionType() {
+        return ProjectionType.MERGE;
+    }
+
+    @Override
+    public <C, R> R accept(ProjectionVisitor<C, R> visitor, C context) {
+        return visitor.visitMergeProjection(this, context);
+    }
+
+    @Override
+    public List<? extends Symbol> outputs() {
+        return outputs;
+    }
+
+    public List<Symbol> orderBy() {
+        return orderBy;
+    }
+
+    public boolean[] reverseFlags() {
+        return reverseFlags;
+    }
+
+    public Boolean[] nullsFirst() {
+        return nullsFirst;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        MergeProjection that = (MergeProjection) o;
+
+        return Objects.equals(orderBy, that.orderBy);
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        int numOutputs = in.readVInt();
+        outputs = new ArrayList<>(numOutputs);
+        for (int i = 0; i < numOutputs; i++) {
+            outputs.add(Symbol.fromStream(in));
+        }
+        int numOrderBy = in.readVInt();
+
+        if (numOrderBy > 0) {
+            reverseFlags = new boolean[numOrderBy];
+
+            for (int i = 0; i < reverseFlags.length; i++) {
+                reverseFlags[i] = in.readBoolean();
+            }
+
+            orderBy = new ArrayList<>(numOrderBy);
+            for (int i = 0; i < reverseFlags.length; i++) {
+                orderBy.add(Symbol.fromStream(in));
+            }
+
+            nullsFirst = new Boolean[numOrderBy];
+            for (int i = 0; i < numOrderBy; i++) {
+                nullsFirst[i] = in.readOptionalBoolean();
+            }
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(outputs.size());
+        for (Symbol symbol : outputs) {
+            Symbol.toStream(symbol, out);
+        }
+        if (isOrdered()) {
+            out.writeVInt(reverseFlags.length);
+            for (boolean reverseFlag : reverseFlags) {
+                out.writeBoolean(reverseFlag);
+            }
+            for (Symbol symbol : orderBy) {
+                Symbol.toStream(symbol, out);
+            }
+            for (Boolean nullFirst : nullsFirst) {
+                out.writeOptionalBoolean(nullFirst);
+            }
+        } else {
+            out.writeVInt(0);
+        }
+
+    }
+
+    public boolean isOrdered() {
+        return reverseFlags != null && reverseFlags.length > 0;
```

Comment:
if reverseFlags is null then the assertion in the constructor would cause an NPE.. so either this check here is not needed or you need to fix the assertion and also mark reverseFlags as nullable